### PR TITLE
Refactor incentive accumulators to be the same

### PR DESCRIPTION
**Pull request by rhuairahrighairidh**
Backports the refactor that was done in adding swap rewards to the existing reward accumulation.

The goal is to increase test coverage of reward accumulation, and standardize behaviour and edge cases across reward types. I also spent a some time on a mini-audit of accumulation to see if I'd missed anything before.

Main Changes
- [x] Standardize on the name "source shares" for the external value this module assigns rewards to (eg swap pool shares, hard bnb supplied, busd-a usdx minted)
- [x] Tweak new accumulation to better handle edge cases (including more tests)
- [x] Backport to borrow rewards
- [x] Backport to supply rewards
- [x] Backport to delegation rewards
- [x] Backport to usdx rewards

### CODE REVIEW

File: x/incentive/spec/01_concepts.md [hunk 1]
```diff
@@ -12,15 +12,15 @@ Rewards target various user activity. For example, usdx borrowed from bnb CDPs,
...
 
 Paying out rewards to every user every block would be slow and lead to long block times. Instead rewards are calculated much less frequently.
 
-Every block a global tracker adds up total rewards paid out per unit of user involvement. For example, per unit of xrpb supplied to hard, or per share in a kava/usdx swap pool. A user's specific reward can then be calculated as needed based on their current deposit/shares/borrow.
+Every block a global tracker adds up total rewards paid out per unit of user involvement. For example, per unit of xrpb supplied to hard, or per share in a kava/usdx swap pool. A user's specific reward can then be calculated as needed based on their current source shares.
 
-User's rewards must be updated whenever their reward source changes. This happens through hooks into other modules that run before deposits/borrows/supplies etc.
+Users' rewards must be updated whenever their source shares change. This happens through hooks into other modules that run before deposits/borrows/supplies etc.
 
 ## HARD Token distribution
```
Comment (karzak) [inline x/incentive/spec/01_concepts.md:25]: Add a TODO to this document that notes we should add a section on SWP token distribution before mainnet launch

**Review by karzak — APPROVED** [review]
tACK - refactor looks good

File: x/incentive/spec/01_concepts.md (hunk 1 above)
Comment (rhuairahrighairidh) [reply x/incentive/spec/01_concepts.md:25]: added in https://github.com/Kava-Labs/kava/pull/970/commits/db91b3862db5c50e3d210353d36e7548abb4cad5
